pkg/signatures/benchmark/signature/golang/anti_debugging.go

```diff
@@ -52,21 +54,22 @@ func (sig *antiDebugging) OnEvent(event protocol.Event) error {
 	if ee.EventName != "ptrace" {
 		return nil
 	}
-	request, err := helpers.GetTraceeArgumentByName(ee, "request", helpers.GetArgOps{DefaultArgs: false})
+	requestArg, err := helpers.GetTraceeIntArgumentByName(ee, "request")
 	if err != nil {
 		return err
 	}
-	requestString := request.Value.(string)
-	if requestString != "PTRACE_TRACEME" {
-		return nil
+
+	if requestArg == int(libbpfgo.PTRACE_TRACEME.Value()) {
+		metadata, err := sig.GetMetadata()
+		if err != nil {
+			return err
+		}
+		sig.cb(detect.Finding{
```

Comment:
I will write it here but it is true for all other signatures.
Should we really omit the data we pass here in the `detect.Finding`?
I think you should parse the argument if the signature is triggered, and pass it through the data member as before.
This way you parse the argument only upon triggering, instead of each event.

Reply:
The most elegant solution I can think of is adding this logic into the detection callback passed in tracee-rules.
However outside of the single binary solution this would mean importing the `events` package into tracee-rules AND tracee-rules needing awareness of parse-arguments configuration (from `OutputConfig`) which is _currently_ a bit of an issue, but when we drop the old arch we can add that.

Otherwise we would need to supply a robust argument parsing solution like we have in `parge_args.go` that is external to tracee's go.mod so that signatures can import it without breaking their plugin compilation (because currently they would have to import `pkg/events`), or let the signature authors take responsibility for the parsing if it matters for their usecase.

Reply:
I agree with you, but as you change the signatures here you are the signature author.
I suggest you use the `libbpfgo` helpers to parse the argument and put the parsed argument in `Data` map.